packages/core/utils/lib/convert-query-params.js

```diff
@@ -185,8 +186,23 @@ const convertPopulateObject = (populate, schema) => {
       return acc;
     }
 
-    // FIXME: This is a temporary solution for dynamic zones that should be
-    // fixed when we'll implement a more accurate way to query them
+    if (typeof subPopulate === 'object' && 'on' in subPopulate) {
```

Comment:
Shouldn't we do this check only for dynamic zone or polymorphic relations ?

Reply:
I would add media to the mix but yes indeed